src/components/confirm/confirm.test.js

```diff
@@ -0,0 +1,386 @@
+
+        if (state === true) {
+          confirmButton().should("be.disabled").and("have.attr", "disabled");
+        } else {
+          confirmButton().should("not.be.disabled");
+        }
+      }
+    );
+
+    it("should check confirm button is loading", () => {
+      CypressMountWithProviders(<ConfirmComponent isLoadingConfirm />);
+
+      confirmButton().should("be.disabled").and("have.attr", "disabled");
+      confirmButton()
+        .children()
+        .children()
+        .children()
+        .should("have.attr", "data-component", "loader")
+        .and("be.visible");
+    });
+
+    it("should check Esc key is disabled", () => {
+      CypressMountWithProviders(<ConfirmComponent disableEscKey />);
+
+      dialogPreview().trigger("keydown", keyCode("Esc"));
+
+      dialogPreview().should("be.visible");
+    });
+
+    it("should check close icon is enabled", () => {
+      CypressMountWithProviders(<ConfirmComponent showCloseIcon />);
+
+      icon().click();
+
+      dialogPreview().should("not.exist");
+    });
+
+    it.each([
+      [true, "rgb(205, 56, 75)", "rgb(205, 56, 75)"],
+      [false, "rgb(0, 126, 69)", "rgb(0, 126, 69)"],
+    ])(
+      "should check cancel button has destructive CSS properties",
+      (state, color, bordercolor) => {
+        CypressMountWithProviders(
+          <ConfirmComponent cancelButtonDestructive={state} />
+        );
+
+        cancelButton()
+          .should("have.css", "color", color)
+          .and("have.css", "border-color", bordercolor);
+      }
+    );
+
+    it.each([
+      [true, "rgb(205, 56, 75)"],
+      [false, "rgb(0, 126, 69)"],
+    ])(
+      "should check confirm button has destructive CSS properties",
+      (state, backgroundcolor) => {
+        CypressMountWithProviders(
+          <ConfirmComponent confirmButtonDestructive={state} />
+        );
+
+        confirmButton().should("have.css", "background-color", backgroundcolor);
+      }
+    );
+
+    it("should check clicking cancel button closes dialog", () => {
+      CypressMountWithProviders(<ConfirmComponent />);
+
+      cancelButton().click();
+      dialogPreview().should("not.exist");
+    });
+
+    it("should check clicking confirm button closes dialog", () => {
+      CypressMountWithProviders(<ConfirmComponent />);
+
+      confirmButton().click();
+
+      dialogPreview().should("not.exist");
+    });
+
+    it("should check clicking confirm button does not close dialog when confirm button is disabled", () => {
+      CypressMountWithProviders(<ConfirmComponent disableConfirm />);
+
+      confirmButton().click({ force: true });
+
+      dialogPreview().should("be.visible");
+    });
+
+    it("should check clicking Esc key does not close dialog when cancel button is disabled", () => {
+      CypressMountWithProviders(<ConfirmComponent disableEscKey />);
+
+      dialogPreview().trigger("keydown", keyCode("Esc"));
```

Comment:
```suggestion
      dialogPreview().trigger("keyup", keyCode("Esc"));
```

Again you could also pass a `cy.stub()` (like you have below) to the `onCancel` prop and assert it isn't called here

Reply:
This test is actually a duplicate of the disableEscKey test above. I've added the cy.stub to that test and replaced this one with a test that checks Esc key does close the confirm dialog.